src/modules/CRUD/services/UserSettingsProvider.js

```diff
@@ -8,20 +8,27 @@ export const TABLE_MINIFIED = 'minified';
 export const TABLE_MEDIUM = 'medium';
 export const TABLE_FULL = 'full';
 
-export const withPageSize = () => connectUserSettingsProvider('pageSize', 'setPageSize');
 export const withTableFilters = () => connectUserSettingsProvider('tableFilters', 'setTableFilters');
 export const withTableSize = () => connectUserSettingsProvider('tableSize', 'setTableSize');
 
 const { Provider } = context;
 
+const getUserSettings = key => (JSON.parse(localStorage.getItem('CRUDUserSettings')) || {})[key];
+
 export const UserSettingsProvider = ({ children }) => {
-  const [pageSize, setPageSize] = useState(10);
-  const [tableSize, setTableSize] = useState(TABLE_MEDIUM);
-  const [tableFilters, setTableFilters] = useState({});
+  const [tableFilters, setTableFilters] = useState(getUserSettings('tableFilters') || {
+    page_size: 10,
+  });
+  const [tableSize, setTableSize] = useState(getUserSettings('tableSize') || TABLE_MEDIUM);
+
+  useEffect(() => {
+    localStorage.setItem('CRUDUserSettings', JSON.stringify({
```

Comment:
I'm not really sure that using a provider pattern for data stored in localStorage is really a good way to do it...

Reply:
The final goal is to get and store these settings via API calls. It's a sort of patch